src/otx/v2/adapters/torch/mmengine/mmpretrain/modules/datasets/otx_datasets.py

```diff
@@ -72,39 +74,35 @@ def get_indices(self, num_classes: list, *args) -> dict:
     def load_annotations(self) -> None:
         """Load annotations."""
         include_empty = self.empty_label in self.labels
-        for i, _ in enumerate(self.otx_dataset):
+        for item in self.otx_dataset:
+            item_labels = [ann.label for ann in item.annotations]
+            ignored_labels = item.attributes.get("ignored_labels", [])
             class_indices = []
-            item_labels = self.otx_dataset[i].get_roi_labels(self.labels, include_empty=include_empty)
-            ignored_labels = self.otx_dataset[i].ignored_labels
             if item_labels:
-                for otx_lbl in item_labels:
-                    if otx_lbl not in ignored_labels:
-                        class_indices.append(self.label_names.index(otx_lbl.name))
+                for label in item_labels:
+                    if label not in ignored_labels:
+                        class_indices.append(label)
                     else:
                         class_indices.append(-1)
             else:  # this supposed to happen only on inference stage
                 class_indices.append(-1)
             self.gt_labels.append(class_indices)
+            self.item_ids.append(item.id)
         self.gt_labels = np.array(self.gt_labels)
 
     def __getitem__(self, index: int) -> dict:
         """Get item from dataset."""
-        dataset = self.otx_dataset
-        item = dataset[index]
-        ignored_labels = np.array([self.label_idx[lbs.id] for lbs in item.ignored_labels])
-
-        height, width = item.height, item.width
-
+        item = self.otx_dataset.get(id=self.item_ids[index])
+        ignored_labels = item.attributes.get("ignored_labels", [])
         gt_label = self.gt_labels[index]
+
         data_info = {
             "dataset_item": item,
-            "width": width,
-            "height": height,
             "index": index,
             "gt_label": gt_label,
             "ignored_labels": ignored_labels,
-            "entity_id": getattr(item, "id_", None),
-            "label_id": self._get_label_id(gt_label),
+            "entity_id": None,
```

Comment:
I remember that @vinnamkim added `DatasetItemEntityWithID` to track per item statistics. Isn't there any identifier in Datumaro item? It's OK if we have other options for tracking.
@vinnamkim What do you think?

Reply:
Can we talk about this when restoring the performance and usage?